drivers/clk/clk-rp1.c

```diff
@@ -1830,9 +1923,25 @@ static const struct rp1_clk_desc clk_desc_array[] = {
 
 	[RP1_CLK_GP0] = REGISTER_CLK(
 				.name = "clk_gp0",
-				.parents = {"xosc"},
+				.parents = {"xosc",
+					    "clksrc_gp1",
+					    "clksrc_gp2",
+					    "clksrc_gp3",
+					    "clksrc_gp4",
+					    "clksrc_gp5",
+					    "pll_sys",
+					    "pll_audio",
+					    "",
```

Comment:
I think the  "-", used elsewhere, looks nicer - could it work instead?

Reply:
The "-" is used as a break/delimiter between the glitchless mux sources and the aux mux sources - the choice of an empty string was deliberate to make the source selection invalid. I'm not sure why the delimiter is necessary - the driver has the information from the num_std_parents and num_aux_parents to work with.

Reply:
There's a gotcha that means a delimiter is needed - the mux assignments are not sequential - the second index into the standard sources is always the aux selector (if a glitchless mux is implemented).